wasm-runtime/support/src/lib.rs

```diff
@@ -0,0 +1,144 @@
+#![no_std]
+#![feature(lang_items)]
+#![feature(alloc)]
+#![cfg_attr(feature = "strict", deny(warnings))]
+
+#![feature(alloc)]
+//#[macro_use]
+extern crate alloc;
+pub use alloc::vec::Vec;
+pub use alloc::boxed::Box;
+pub use alloc::rc::Rc;
+pub use core::mem::{transmute, size_of, uninitialized, swap};
+pub use core::slice;
+pub use core::cell::{RefCell, Ref, RefMut};
+
+extern crate pwasm_libc;
+extern crate pwasm_alloc;
+
+#[lang = "panic_fmt"]
+#[no_mangle]
+pub fn panic_fmt() -> ! {
+	  loop {}
+}
+
+extern "C" {
+	fn ext_print(utf8_data: *const u8, utf8_len: u32);
+	fn ext_print_num(value: u64);
+	fn ext_set_storage(key_data: *const u8, key_len: u32, value_data: *const u8, value_len: u32);
+	fn ext_get_allocated_storage(key_data: *const u8, key_len: u32, written_out: *mut u32) -> *mut u8;
+	fn ext_get_storage_into(key_data: *const u8, key_len: u32, value_data: *mut u8, value_len: u32) -> u32;
+	fn ext_chain_id() -> u64;
+	fn ext_blake2_256(data: *const u8, len: u32, out: *mut u8);
+	fn ext_twox_128(data: *const u8, len: u32, out: *mut u8);
```

Comment:
What's the point of extending this two 128 and 256 bits? Since it is not crypto-secure I'd expect the basic 32-bit or 64-bit variant to be the most useful.

Reply:
yeah primarily to avoid accidental clashes - it might not be crypto-secure but we can still use it when we're sure that there won't be the possibility of targeted preimaging attacks.